DigitalOffice/Tests/AuthenticationServiceTests/Broker/Sender/AuthenticationSenderTests.cs

```diff
@@ -0,0 +1,103 @@
+    {
+        public bool IsSuccess { get; set; }
+
+        public List<string> Errors { get; set; }
+
+        public T Body { get; set; }
+    }
+
+    public class UserCredentialsRequest : IUserCredentialsRequest
+    {
+        public string Email { get; set; }
+    }
+
+    public class UserCredentialsResponse : IUserCredentialsResponse
+    {
+        public Guid UserId { get; set; }
+        public string PasswordHash { get; set; }
+    }
+
+    class AuthenticationSenderTests
+    {
+        private Mock<Response<IOperationResult<IUserCredentialsResponse>>> responseBrokerMock;
+        private Mock<IRequestClient<IUserCredentialsRequest>> requestBrokerMock;
+        private OperationResult<UserCredentialsResponse> operationResult;
+        private UserCredentialsResponse brokerResponse;
+        private AuthenticationSender authSender;
+
+        [SetUp]
+        public void SetUp()
+        {
+            responseBrokerMock = new Mock<Response<IOperationResult<IUserCredentialsResponse>>>();
+            requestBrokerMock = new Mock<IRequestClient<IUserCredentialsRequest>>();
+            authSender = new AuthenticationSender(requestBrokerMock.Object);
+
+            var brokerRequest = new UserCredentialsRequest
+            {
+                Email = "Example@gmail.com"
+            };
+
+            brokerResponse = new UserCredentialsResponse
+            {
+                UserId = Guid.NewGuid(),
+                PasswordHash = "Example"
+            };
+
+            operationResult = new OperationResult<UserCredentialsResponse>
+            {
+                IsSuccess = true,
+                Errors = new List<string>(),
+                Body = brokerResponse
+            };
+
+            requestBrokerMock.Setup(
+                x => x.GetResponse<IOperationResult<IUserCredentialsResponse>>(
+                    It.IsAny<object>(), default, default))
+                .Returns(Task.FromResult(responseBrokerMock.Object));
+        }
+
+        [Test]
+        public void ShouldReturnUserIdAndPasswordWhenEmailHashMatch()
+        {
+            var userEmail = "Example@gmail.com";
+
+            responseBrokerMock
+                .SetupGet(x => x.Message)
+                .Returns(operationResult);
+
+            SerializerAssert.AreEqual(brokerResponse, authSender.GetResponseFromBroker(userEmail).Result);
+        }
+
+        [Test]
+        public void ShouldThrowExceptionWhenEmailHashNotMatch()
+        {
+            var userEmail = "Example@gmail.com";
+            var expectedErrors = new List<string>() { "User email not found" };
+
+            operationResult.IsSuccess = false;
+            operationResult.Errors = new List<string>() { "User email not found" };
+            operationResult.Body = null;
+
+            responseBrokerMock
+                .SetupGet(x => x.Message)
+                .Returns(operationResult);
+
+            Assert.ThrowsAsync<Exception>(() => authSender.GetResponseFromBroker(userEmail));
```

Comment:
Should change to `Assert.That(() => authSender.GetResponseFromBroker(userEmail),
                Throws.InstanceOf<Exception>().And.Message.EqualTo("User email not found"));`

Reply:
Fixed